lib/std/unicode.zig

```diff
@@ -770,6 +769,48 @@ fn calcUtf16LeLen(utf8: []const u8) usize {
     return dest_len;
 }
 
+/// Print the given `utf16le` string
+fn formatUtf16le(
+    utf16le: []const u16,
+    comptime fmt: []const u8,
+    options: std.fmt.FormatOptions,
+    writer: anytype,
+) !void {
+    const unknown_codepoint = 0xfffd;
+    _ = fmt;
+    _ = options;
+    var buf: [300]u8 = undefined; // just a random size I chose
+    var it = Utf16LeIterator.init(utf16le);
+    var u8len: usize = 0;
+    while (it.nextCodepoint() catch unknown_codepoint) |codepoint| {
+        u8len += utf8Encode(codepoint, buf[u8len..]) catch
+            utf8Encode(unknown_codepoint, buf[u8len..]) catch unreachable;
+        if (u8len + 3 >= buf.len) {
+            try writer.writeAll(buf[0..u8len]);
+            u8len = 0;
+        }
+    }
+    try writer.writeAll(buf[0..u8len]);
+}
+
+/// Return a Formatter for a Utf16le string
+pub fn fmtUtf16le(utf16le: []const u16) std.fmt.Formatter(formatUtf16le) {
+    return .{ .data = utf16le };
+}
+
+test "fmtUtf16le" {
+    const expectFmt = std.testing.expectFmt;
+    try expectFmt("", "{}", .{fmtUtf16le(utf8ToUtf16LeStringLiteral(""))});
+    try expectFmt("foo", "{}", .{fmtUtf16le(utf8ToUtf16LeStringLiteral("foo"))});
+    try expectFmt("𐐷", "{}", .{fmtUtf16le(utf8ToUtf16LeStringLiteral("𐐷"))});
+    try expectFmt("퟿", "{}", .{fmtUtf16le(&[_]u16{std.mem.readIntNative(u16, "\xff\xd7")})});
```

Comment:
Does `'\u{ffd7}'` not work here?

Reply:
It would work here, but it wouldn't work for the lines below because they are invalid unicode byte sequences, and the point of this one and the last one are to be the previous/next byte around the invalid range, so I used the same format to make that clear.

Reply:
I don't think we validate these unicode codepoints anywhere as https://godbolt.org/z/r6nd8Ybfh compiles just fine.

Reply:
You need to reverse the byte order.  Try `"\u{d800}"` and you should get the error.

Reply:
Right, still not seeing errors https://godbolt.org/z/q56eKfaKo. Probably would be good to add errors here though, Rust at least seems to give an error on `"\u{d800}"`.

Reply:
you get an error when you pass it to `utf8ToUtf16LeStringLiteral` like you would do to use it in this test block

Reply:
But why would you do that when it's just one character?

I guess this is fine too since it's just some test but generally I'd prefer `"\xXX"` patterns to be the last resort.

Reply:
The order of the bytes must change depending on the platform endianness, and we can't use the utf8 decoder functions to decode them because they are invalid, so I'm specifying the bytes in little endian and using the native integer decoding.  Note that before I just had the bytes in the same order but it failed on a big endian platform in the CI so I added the rest to fix it.

If you have a simpler way to make this work please share :)

Reply:
Sounds like a bitcasted array of `u8` would also work, but I might just be confusing myself with all the endianness .